### PR TITLE
Generating .properties files that are placed in jars.

### PR DESCRIPTION
**Pull request by albsadowski**

### CODE REVIEW

**Comment by gkossakowski** [comment]
Hi @dumpstate!
Thanks for the PR! Is this a direct port of what Ant does at the moment?

**Comment by albsadowski** [comment]
I've tried to port the behavior of Ant. But I've just noticed that I'm missing the build.release property that is changing the version.number. I'll fix this and setup another PR.

At the ScalaWaw, we talked about using sbt-git or jgit to retrieve the git commit date and sha, but just after the meeting I've checked that the Ant is using the scripts provided in the tools directory. What do you think about this approach?

**Comment by gkossakowski** [comment]
I think it's a good idea to follow what Ant does as a first step. Please just mention this in the code and commit message so we know where this came from.

File: build.sbt [hunk 1]
```diff
@@ -89,7 +89,47 @@ lazy val commonSettings = Seq(
...
+  copyrightString := "Copyright 2002-2013, LAMP/EPFL",
+
+  resourceGenerators in Compile += generateVersionPropertiesFile.taskValue,
+
+  generateVersionPropertiesFile := {
+    val propFile = (resourceManaged in Compile).value / s"${name.value}.properties"
+    val props = new java.util.Properties
+
+    val (ver, mavenVer, osgiVer) =
+      version.value.split("-").toList match {
```
Comment (gkossakowski) [inline build.sbt:103]: Have you considered expressing this logic using regexp matchers? You could even inline regexps into patterns using string interpolation if that helps readability. Check out this article for details: https://github.com/OlegIlyenko/hacking-scala-blog/blob/master/posts/String-Interpolation-Meets-Regular-Expressions.md

**Comment by gkossakowski** [comment]
I'll grab this one and look into using regexps myself. Thanks for the PR!

**Pull request merged**